File: backend/src/backend/schema/comment.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class CommentCreate(BaseModel):
    blog_id: int
    content: str
    reply_to_id: Optional[int] = None


class Comment(BaseModel):
    id: int
    content: str
    created_at: datetime
    updated_at: datetime
    user_id: int
    blog_id: int
    reply_to_id: Optional[int] = None

    class Config:
        from_attributes = True


class CommentsResponse(BaseModel):
    comments: List[Comment] = []
    total: int = 0

    class Config:
        from_attributes = True

File: frontend/types/comment.ts
export interface Comment {
  id: number;
  blog_id: number;
  user_id: number;
  content: string;
  reply_to_id: number | null;
  created_at: number;
  updated_at: number;
  user?: CommentUser;
  replies?: Comment[];
}

export interface CommentUser {
  uuid: string;
  username: string;
  avatar_url: string;
}

export interface CommentCreate {
  blog_id: number;
  content: string;
  reply_to_id?: number | null;
}

export interface CommentsResponse {
  comments: Comment[];
  total: number;
}
